src/pyqasm/visitor.py

```diff
@@ -352,6 +360,15 @@ def _visit_quantum_register(
 
         self._module._add_qubit_register(register_name, register_size)
 
+        # Inline: Update offsets after adding a new register if device_qubits is set
+        if self._consolidate_qubits:
+            offsets = {}
```

Comment:
We don't need to construct this repeatedly right? Just get the current max offset, add it as the offset of the current qubit register and then increment the max offset by current register size

Also, we need to preserve the ordering of the qubits appearing in our quantum program - using an [ordered dict](https://docs.python.org/3/library/collections.html#collections.OrderedDict) should help keep the order of the qubit declarations.